plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -625,7 +626,7 @@ public void setTableComment(ConnectorSession session, ConnectorTableHandle table
     public Optional<ConnectorTableLayout> getNewTableLayout(ConnectorSession session, ConnectorTableMetadata tableMetadata)
     {
         Schema schema = toIcebergSchema(tableMetadata.getColumns());
-        PartitionSpec partitionSpec = parsePartitionFields(schema, getPartitioning(tableMetadata.getProperties()));
+        PartitionSpec partitionSpec = createPartitionSpec(schema, getPartitioning(tableMetadata.getProperties()));
```

Comment:
When should i call `createPartitionSpec` and when `parsePartitionFields`?
they have similar names, and even more similar semantics.

Also, how does introduction of the wrapper (that throws TrinoException) related to adding quoted identifiers?
the parsing could fail even before the change, right? (eg unsupported transform, missing closing brace, etc)

Reply:
The exceptions caught would be related to the parsing of the partitioning field, translating the iceberg mini language into an Iceberg spec. 

As we got here it means the SQL statement is valid. Here we only validate the string content of the PARTITIONING field.

The validation of the user input is expected error checking that's why I think they should be Trino exceptions. Maybe a good point would be that we should not be able to directly call `parsePartitionFields`, make it package private and move `createPartitionSpec` into this class.

Reply:
I removed the wrapper and just catch the exception in `parsePartitionFields` now and rethrow as `TrinoException`. This makes the testing easier in `BaseConnectorTest` as we are expecting `TrinoException` there,